RestfulDropApi.cs

```diff
@@ -0,0 +1,124 @@
+                {
+                     manifestResponse = await client.GetAsync(manifestResponse.Headers.Location);
+                }
+                manifestResponse.EnsureSuccessStatusCode();
+                string manifestContent = await manifestResponse.Content.ReadAsStringAsync();
+                     
+                //filter here so we can be case insensitve. manifest url would take a directory but unlike root in drop.exe it is case sensitve.
+                var manifest = JsonConvert.DeserializeObject<List<VstsFile>>(manifestContent)
+                                    .Where(f => f.Path.StartsWith(relativeRoot, StringComparison.OrdinalIgnoreCase));
+                
+                
+                // forget what our limit was but this would be bad for a whole drop. Batch has a certain
+                // limit. Should we batch to 500 or lazily get sas urls for only certain directories
+                // (those with dockerfiles in them)
+                var uniqueBlobs = manifest.Select(file => file.Blob.Id).Distinct().Select(id => new { id =  id});
+                var blobList = JsonConvert.SerializeObject(new { blobs = uniqueBlobs });
+                var content = new StringContent(blobList);
+                content.Headers.ContentType = MediaTypeHeaderValue.Parse("application/json");
+
+                var blobHostname = manifestUri.Host.Replace("artifacts", "vsblob");
+                var uriBuilder = new UriBuilder(manifestUri.Scheme, blobHostname, -1, "defaultcollection/_apis/blob/blobsbatch");
+                var queryParameters = HttpUtility.ParseQueryString("");
+                queryParameters.Add(APIVersionParam, blobAPIVersion);
+                uriBuilder.Query = queryParameters.ToString();
+                
+                _logger.LogDebug($"asking for sas tokens at {uriBuilder.Uri}");
+                var blobRequest = new HttpRequestMessage(HttpMethod.Post, uriBuilder.Uri);
+                blobRequest.Headers.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Basic", base64EncodedString);
+                blobRequest.Content = content;
+                var response = await client.SendAsync(blobRequest);
+                response.EnsureSuccessStatusCode();
+
+                string sasUrlJson = await response.Content.ReadAsStringAsync();
+                var blobBatch = JsonConvert.DeserializeObject<BatchBlobResponse>(sasUrlJson);
+                
+                var urlDictionary = blobBatch.Blobs.ToDictionary(b => b.Id, b => b.Url);
+                foreach (var file in manifest)
+                {
+                    file.Blob.Url = urlDictionary[file.Blob.Id];
+                }
+                return manifest;
+            }
+        }   
+
+        internal sealed class BatchBlobResponse
+        {
+            /// <summary>
+            /// Blob ids and SAS urls.
+            /// </summary>
+            #pragma warning disable 0649
```

Comment:
don't remember this pragma.

Reply:
This was the only question I actualy had for chip. The rest I can actually clean up.

Reply:
I was going to ask you about the #pragma as well, wasn't sure why it was there. Can chat tomorrow.

Reply:
Never assigend. which is wrong it's assigned by DeserializeObject

Reply:
-            #pragma warning disable 0649
-            public IEnumerable<VstsBlob> Blobs;
-            #pragma warning restore 0169
+
+            public IEnumerable<VstsBlob> Blobs { get; set; }